sdk/communication/azure-communication-sms/tests/test_sms_client_e2e.py

```diff
@@ -60,45 +70,100 @@ def test_send_sms_multiple(self):
 
-        for sms_response in sms_responses:
-            assert sms_response.http_status_code == 400
-            assert not sms_response.successful
+        self.verify_successful_sms_response(sms_responses[0])
+
+    @pytest.mark.live_test_only
+    def test_send_sms_fake_from_phone_number(self):
+
+        sms_client = SmsClient.from_connection_string(self.connection_str)
+        
+        with pytest.raises(HttpResponseError) as ex:
+            # calling send() with sms values
+            sms_client.send(
+                from_="+15550000000",
+                to=[self.phone_number],
+                message="Hello World via SMS")
+        
+        assert str(ex.value.status_code) == "400"
+        assert ex.value.message is not None
+
+    @pytest.mark.live_test_only
+    def test_send_sms_fake_to_phone_number(self):
+
+        sms_client = SmsClient.from_connection_string(self.connection_str)
+
+        # calling send() with sms values
+        sms_responses = sms_client.send(
+            from_=self.phone_number,
+            to=["+15550000000"],
+            message="Hello World via SMS")
+        
+        assert len(sms_responses) is 1
+
+        assert sms_responses[0].message_id is None
+        assert sms_responses[0].http_status_code == 400
+        assert sms_responses[0].error_message == "Invalid To phone number format."
+        assert not sms_responses[0].successful
     
+    @pytest.mark.live_test_only
+    def test_send_sms_unauthorized_from_phone_number(self):
+
+        sms_client = SmsClient.from_connection_string(self.connection_str)
+        
+        with pytest.raises(HttpResponseError) as ex:
+            # calling send() with sms values
+            sms_client.send(
+                from_="+18007342577",
+                to=[self.phone_number],
+                message="Hello World via SMS")
+        
+        assert str(ex.value.status_code) == "404"
+        assert ex.value.message is not None
+
     @pytest.mark.live_test_only
     def test_send_sms_unique_message_ids(self):
 
+        sms_client = SmsClient.from_connection_string(self.connection_str)
+
         # calling send() with sms values
-        sms_responses_1 = self.sms_client.send(
+        sms_responses_1 = sms_client.send(
             from_=self.phone_number,
             to=[self.phone_number],
             message="Hello World via SMS")
         
         # calling send() again with the same sms values
-        sms_responses_2 = self.sms_client.send(
+        sms_responses_2 = sms_client.send(
             from_=self.phone_number,
             to=[self.phone_number],
             message="Hello World via SMS")
         
+        self.verify_successful_sms_response(sms_responses_1[0])
+        self.verify_successful_sms_response(sms_responses_2[0])
         # message ids should be unique due to having a different idempotency key
         assert sms_responses_1[0].message_id != sms_responses_2[0].message_id
```

Comment:
any idea why this works in playback mode even though it's sanitized in the recording?

Reply:
This test is only enabled for live mode.

Reply:
oh interesting! i wonder if i should do that with JS